distribution_metadata.c

```diff
@@ -829,22 +744,15 @@ LockShard(int64 shardId, LOCKMODE lockMode)
 	/* locks use 32-bit identifier fields, so split shardId */
 	uint32 keyUpperHalf = (uint32) (shardId >> 32);
 	uint32 keyLowerHalf = (uint32) shardId;
+	bool sessionLock = false;   /* we want a transaction lock */
+	bool dontWait = false;      /* block indefinitely until acquired */
 
 	LOCKTAG lockTag;
 	memset(&lockTag, 0, sizeof(LOCKTAG));
 
-	SET_LOCKTAG_ADVISORY(lockTag, MyDatabaseId, keyUpperHalf, keyLowerHalf, 0);
+	Assert(lockMode == ExclusiveLock || lockMode == ShareLock);
```

Comment:
Is this change related to the SPI one? Any reason to convert it from an error to Assert?